### PR TITLE
Neo4j configure auth

### PR DESCRIPTION
**Pull request by mneedham**
This PR makes auth configurable and sets the default to enabled rather than disabled. 

Also updated the notes to explain how to set a new password.

### CODE REVIEW

File: incubator/neo4j/templates/NOTES.txt [hunk 1]
```diff
@@ -1,9 +1,35 @@
-You can connect to the cluster using the Cypher shell by running:
+We'll need to wait a few seconds for the Neo4j cluster to form.
+We need to see this line in all of our pods' logs:
+
+> Remote interface available at http://neo-helm-neo4j-core-2.neo-helm-neo4j.default.svc.cluster.local:7474/
+
+We can see the content of the logs by running the following command:
+
+kubectl logs -l "app=neo4j,component=core"
```
Comment (unguiculus) [inline incubator/neo4j/templates/NOTES.txt]: Don't hard-code the app label:
-> `kubectl logs -l "app={{ template "neo4j.name" . }},component=core"`

File: incubator/neo4j/templates/tests/test-config.yaml [hunk 2]
```diff
@@ -15,10 +15,12 @@ data:
         name=$1
         end="$((SECONDS+90))"
         while true; do
-          echo "testing http://$name:7474/db/data"
-          out=`wget http://$name:7474/db/data -O - 2>&1`
+          kubectl exec $name -- bin/cypher-shell -u neo4j -p neo4j "CALL dbms.security.changePassword('newPassword')"
```
Comment (unguiculus) [inline incubator/neo4j/templates/tests/test-config.yaml]: Tests should not change the password.

**Review by unguiculus — CHANGES_REQUESTED** [review]
It should be possible to specify or auto-generate the password instead of having to change it after the installation. Tests should then use this password instead of changing it. Use a secret for the password. Here's an example: https://github.com/kubernetes/charts/blob/master/stable/mariadb/templates/secrets.yaml

**Comment by mneedham** [comment]
@unguiculus just got around to addressing your suggestions. I've now implemented something similar to the Maria DB one with auto generated password stored in secrets. 

Let me know if there's anything else I should change.

**Comment by mneedham** [comment]
@unguiculus Pinging to see if there's anything else I need to do :)

File: incubator/neo4j/templates/secrets.yml [hunk 3]
```diff
@@ -0,0 +1,18 @@
+{{- if .Values.authEnabled -}}
+apiVersion: v1
+kind: Secret
+metadata:
+  name: {{ template "neo4j.secrets.fullname" . }}
+  labels:
+    app: {{ template "neo4j.secrets.fullname" . }}
```
Comment (unguiculus) [inline incubator/neo4j/templates/secrets.yml]: `app: {{ template "neo4j.name" . }}`

File: incubator/neo4j/templates/NOTES.txt [hunk 4]
```diff
@@ -1,9 +1,25 @@
...
+> Remote interface available at http://neo-helm-neo4j-core-2.neo-helm-neo4j.default.svc.cluster.local:7474/
+
+We can see the content of the logs by running the following command:
+
+kubectl logs -l "app={{ template "neo4j.name" . }},component=core"
+
+We can now run a query to find the topology of the cluster.
+
+kubectl run -it --rm cypher-shell \
+    --image=neo4j:3.2.1-enterprise \
```
Comment (unguiculus) [inline incubator/neo4j/templates/NOTES.txt]: `imageTag` is `3.2.3`

**Review by unguiculus — CHANGES_REQUESTED** [review]
Sorry for the delay. We're almost there.

**Comment by mneedham** [comment]
@unguiculus done!

File: incubator/neo4j/templates/secrets.yml [hunk 5]
```diff
@@ -0,0 +1,18 @@
+{{- if .Values.authEnabled -}}
+apiVersion: v1
+kind: Secret
+metadata:
+  name: app: {{ template "neo4j.name" . }}
+  labels:
+    app: app: {{ template "neo4j.name" . }}
```
Comment (unguiculus) [inline incubator/neo4j/templates/secrets.yml]: One `app:` too many

File: incubator/neo4j/templates/secrets.yml [hunk 6]
```diff
@@ -0,0 +1,18 @@
+{{- if .Values.authEnabled -}}
+apiVersion: v1
+kind: Secret
+metadata:
+  name: app: {{ template "neo4j.name" . }}
```
Comment (unguiculus) [inline incubator/neo4j/templates/secrets.yml]: You should have only changed the `app` label, not the name. Please change it back to `{{ template "neo4j.secrets.fullname" . }}`

**Review by unguiculus — CHANGES_REQUESTED** [review]
Please don't squash commits when updating the PR. Github can't show the changes then. The PR will be squashed anyway when it is merged.

File: incubator/neo4j/templates/secrets.yml (hunk 5 above)
Comment (mneedham) [reply incubator/neo4j/templates/secrets.yml]: @unguiculus ooops, sorry. Have tried again.

**Review by unguiculus — APPROVED** [review]

**Comment by unguiculus** [comment]
I installed the chart successfully and ran tests on my GKE cluster. Tests seems to never finish. Here's the test log I get:

```
$ stern misty-bat-neo4j-service-test-1xcb3                                                                                      ± ★ ✎ ⫘ (97e8557)
+ misty-bat-neo4j-service-test-1xcb3 › misty-bat-ui-test
misty-bat-neo4j-service-test-1xcb3 misty-bat-ui-test 1..2
misty-bat-neo4j-service-test-1xcb3 misty-bat-ui-test ok 1 Testing Neo4j cluster has quorum
misty-bat-neo4j-service-test-1xcb3 misty-bat-ui-test ok 2 Testing we can get the cluster role of each server
```

**Comment by unguiculus** [comment]
@mneedham Did you get a chance to look at the tests?

**Comment by mneedham** [comment]
@unguiculus I've just tried it locally using these commands:

```
$ helm init && kubectl rollout status -w deployment/tiller-deploy --namespace=kube-system
$ helm install incubator/neo4j --wait --name neo-helm

$ helm test neo-helm
RUNNING: neo-helm-neo4j-service-test-gmp9b
PASSED: neo-helm-neo4j-service-test-gmp9b
```

And it seems happy. I have similar logs to the ones you're seeing:

```
$ kubectl logs neo-helm-neo4j-service-test-gmp9b
1..2
ok 1 Testing Neo4j cluster has quorum
ok 2 Testing we can get the cluster role of each server
```

**Comment by mneedham** [comment]
The lint checker is unhappy with the yml prefix so I've renamed the secret file

**Comment by unguiculus** [comment]
Please update to latest master. This will probably fix the linting issue.

**Comment by mneedham** [comment]
I've rebased against master

**Comment by unguiculus** [comment]
I ran Helm tests successfully on my cluster. CI probably just times out.

**Comment by unguiculus** [comment]
I think you could move the chart to stable.

**Pull request merged**